### ISSUE TITLE
Through relations not saving data to new objects when related model is not created under its own api

### ISSUE THREAD

**Issue by hellsan631** [strongloop/loopback#1278]
I've been writing a lot of tests via supertest to make sure my loopback relations are working as intended, when I came across this possible bug. I don't know if i'm doing something incorrectly, but data is not being saved, which makes me believe that this is a bug.

Basically, when you add a model relation of hasManyThrough you get access to some added API calls.

I have 3 models here; Person, Meetup, and Invite
```
Person hasMany Meetups Through Invite
Meetup hasMany Invites
Invite belongsTo Person
invite belongsTo Meetup
```

I get access to these API's because of the relations between them:
```
/people/{id}/meetups
/people/{id}/meetups/rel/{fk}
```

Now when I create two people, and create the meetup through "POST to /people/{id}/meetups", It doesn't want to save any data when I add a related invite using that same persons ID

But if I create the meetup via "POST to /meetups" everything works correctly.

Here is a travis log that better demonstrates the issue:
https://travis-ci.org/hellsan631/relations-through-test

And accompanying Git repo
https://github.com/hellsan631/relations-through-test

**Comment by loay** [comment]
Hi @hellsan631 , 
I tested your repo and I tested the failed cases using explorer and they worked fine, so it is probably something with the test cases.  I see the same test cases failing as well on my end, however they did work fine on explorer. I tried only the first 2. 

Also, I don't see a problem with POST in the first 2 cases, it is in the PUT and GET. 
Let me know if you are still running into the same issue. Thanks.
